code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -817,6 +817,52 @@
 	log_game("DYNAMIC: [key_name(S)] was spawned as a Morph by the midround ruleset.")
 	return S
 
+
+//////////////////////////////////////////////
+//                                          //
+//           PRISONER  (GHOST)            	//
+//                                          //
+//////////////////////////////////////////////
+/datum/dynamic_ruleset/midround/from_ghosts/prisoners
+	name = "Prisoners"
+	midround_ruleset_style = MIDROUND_RULESET_STYLE_LIGHT
+	role_preference = /datum/role_preference/midround_ghost/prisoner
+	required_type = /mob/dead/observer
+	enemy_roles = list(JOB_NAME_SECURITYOFFICER, JOB_NAME_WARDEN, JOB_NAME_HEADOFSECURITY)
+	required_enemies = list(3,1,1,0,0,0,0,0,0,0)
+	required_candidates = 1
+	weight = 2
+	cost = 6
+	minimum_players = 20
+	repeatable = FALSE
+	var/list/spawn_locs
+
+/datum/dynamic_ruleset/midround/from_ghosts/prisoners/acceptable(population=0, threat=0)
+	if (!SSmapping.empty_space)
+		return FALSE
+	return ..()
+
+/datum/dynamic_ruleset/midround/from_ghosts/prisoners/ready(forced = FALSE)
+	if(!..())
+		return FALSE
+	spawn_locs = list()
+	for(var/obj/effect/landmark/prisonspawn/L in GLOB.landmarks_list)
+		if(isturf(L.loc))
+			spawn_locs += L.loc
+	if(!length(spawn_locs))
+		log_game("DYNAMIC: [ruletype] ruleset [name] ready() failed due to no valid spawn locations.")
+		return FALSE
+	return TRUE
+
+/datum/dynamic_ruleset/midround/from_ghosts/prisoners/review_applications()
+	var/turf/landing_turf = pick(spawn_locs)
+	var/result = spawn_prisoners(landing_turf, candidates, list())
```

Comment:
spawning of prisoners should probably be done in `generate_ruleset_body` with the parent of `review_applications` taking care of enough applicants and such. Only gripe I could find with this PR.

Reply:
Errr sure! I'll take a look this night!

Reply:
I'm currently doing the same thing that Fugitives do 
https://github.com/BeeStation/BeeStation-Hornet/blob/bbb6a24b339b3457f45b9f729b3b2b9d82190a4f/code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm#L861-L868


I think I could do that, but I'd have to change a lot of the pr for a style change!

Reply:
Discussed in discord, it's fine as is.